dex-it-common/src/main/scala/com/wavesplatform/dex/it/docker/DexContainer.scala

```diff
@@ -0,0 +1,109 @@
+            runConfig: Config,
+            suiteInitialConfig: Config,
+            localLogsDir: Path)(implicit
+                                tryHttpBackend: LoggingSttpBackend[Try, Nothing],
+                                futureHttpBackend: LoggingSttpBackend[Future, Nothing],
+                                ec: ExecutionContext): DexContainer = {
+
+    val underlying = GenericContainer(
+      dockerImage = "com.wavesplatform/dex-it:latest",
+      exposedPorts = Seq(restApiPort),
+      env = getEnv(name),
+      waitStrategy = ignoreWaitStrategy
+    ).configure { c =>
+      c.withNetwork(network)
+      c.withNetworkAliases("d3x")
+      c.withFileSystemBind(localLogsDir.toString, containerLogsPath, BindMode.READ_WRITE)
+      c.withCreateContainerCmdModifier {
+        _.withName(s"$networkName-$name") // network.getName returns random id
+          .withIpv4Address(internalIp): Unit
+      }
+
+      // Copy files to container
+      List(
+        ("dex-base.conf", getRawContentFromResource(s"dex-servers/dex-base.conf"), false),
+        (s"$name.conf", getRawContentFromResource(s"dex-servers/$name.conf"), false),
+        ("run.conf", runConfig.rendered, true),
+        ("suite.conf", suiteInitialConfig.rendered, true),
+        ("/doc/logback-container.xml", getRawContentFromResource("dex-servers/logback-container.xml"), false)
+      ).map {
+        case (fileName, content, logContent) =>
+          val containerPath = Paths.get(baseContainerPath, fileName).toString
+          log.trace(s"[name=$name] Write to '$containerPath'${if (logContent) s":\n$content" else ""}")
+          c.withCopyFileToContainer(MountableFileOps.fromContent(content), containerPath)
+      }
+    }
+
+    DexContainer(internalIp, underlying)
+  }
+
+  private def getEnv(containerName: String): Map[String, String] = Map(
+    "WAVES_DEX_DETAILED_LOG_PATH" -> s"$containerLogsPath/container-$containerName.detailed.log",
+    "WAVES_DEX_CONFIGPATH"        -> s"$baseContainerPath/$containerName.conf",
+    "WAVES_DEX_OPTS" ->
+      List(
+        "-J-Xmx1024M",
+        "-Dlogback.stdout.enabled=false",
+        "-Dlogback.file.enabled=false",
+        s"-Dlogback.configurationFile=$baseContainerPath/doc/logback.xml",
+        s"-Dlogback.include.file=$baseContainerPath/doc/logback-container.xml",
+        s"-Dlogback.brief.fullPath=$containerLogsPath/container-$containerName.log",
+        s"-Dlogback.detailed.fullPath=$containerLogsPath/container-$containerName.detailed.log"
+      ).prependIf(isProfilingEnabled) {
+          // https://www.yourkit.com/docs/java/help/startup_options.jsp
+          s"-J-agentpath:/usr/local/YourKit-JavaProfiler-2019.8/bin/linux-x86-64/libyjpagent.so=port=10001,listen=all" +
+            s",sampling,monitors,sessionname=prof-$containerName,snapshot_name_format={sessionname}," +
+            s"dir=$containerLogsPath,logdir=$containerLogsPath,onexit=snapshot"
+        }
+        .mkString(" ", " ", " ")
+  )
```

Comment:
Why did you extract this code from _BaseContainer_ (former `create` method) and in fact duplicate it in _DexContainer_ and _WavesNodeContainer_?

Reply:
1. It is likely took too much efforts to maintain a common method. It contains a common code, but most arguments to them are differ. Also the code was simple here.

2. It was uncomfortable to edit container's starting options without seeing environment variables (and modifying it too!). This is the reason I moved content of *ContainerInfo classes to related companion objects. There are also a golden rule: keep entities those likely change together in same place. I believe we don't follow it everywhere, but it is a good rule, let's aim it

3. Not big deal, but with different special "create" methods for both containers we get the freedom from any Node's changes. Probably we will use their docker image (from docker hub) in the future. This change won't require a new round of code generalization.